src/DotLiquid/RenderParameters.cs

```diff
@@ -107,15 +113,15 @@ internal void Evaluate(Template template, out Context context, out Hash register
                 environments.Add(LocalVariables);
             if (template.IsThreadSafe)
             {
-                context = new Context(environments, new Hash(), new Hash(), ErrorsOutputMode, MaxIterations, Timeout, FormatProvider)
+                context = new Context(environments, new Hash(), new Hash(), ErrorsOutputMode, MaxIterations, FormatProvider, Timeout, CancellationToken)

```

Comment:
Not clear how this is either or. True this constructor is internal, but here you exposed the possibility of having both a timeout and a cancellation token. Respecting both, as your commit message says, convolutes things even more.